api/tests/tests_messages.py

```diff
@@ -0,0 +1,70 @@
+from django.test import TestCase
+from django.urls import reverse
+from api.models import Conversation, Task, User
+import json
+
+# Message-related endpoint tests reside here.
+
+
+
+# This tests createtask, createconversation and postmessage.
+class PostMessageTests(TestCase):
+    def setUp(self):
+        self.client.generic('GET',
+                        reverse('api:signup'),
+                        json.dumps(
+                        {
+                            "name": "Jon Doe",
+                            "email": "jon@example.com",
+                            "oauthid": "abcdefg",
+                            "photourl": "https://google.com/test.png",
+                            "location": "Philadelphia",
+                            "bio": "Hello.",
+                        }),
+                        content_type='application/json')
+
+        self.client.generic('GET',
+                            reverse('api:login'),
+                            json.dumps(
+                            {
+                                "email": "jon@example.com",
+                                "oauthid": "abcdefg"
+                            }),
+                            content_type='application/json')
+
+        self.client.generic('GET',
+                            reverse('api:createtask'),
+                            json.dumps(
+                            {
+                                "description": "This is a test task",
+                            }),
+                            content_type='application/json')
+
+        self.user = User.objects.get(email="jon@example.com")
+        self.task = Task.objects.get(description="This is a test task")
+
+        self.client.generic('GET',
+                            reverse('api:createconversation'),
+                            json.dumps(
+                            {
+                                "pinid": str(self.user.id),
+                                "volunteerid": str(self.user.id),
+                                "taskid": str(self.task.id),
+                            }),
+                            content_type='application/json')
```

Comment:
This makes sure that every object is created and saved in DB since if one of them fails that 1 test case fails. I created this flow cause I needed some conversation ID which has some form that I don't know.